staging/src/k8s.io/kubelet/config/v1beta1/types.go

```diff
@@ -851,6 +851,10 @@ type KubeletConfiguration struct {
 	// If not specified, the value in containerRuntimeEndpoint is used.
 	// +optional
 	ImageServiceEndpoint string `json:"imageServiceEndpoint,omitempty"`
+
+	// PullImageSecretRecheckPeriod defines the duration to recheck the pull image secret.
+	// By default, the kubelet will recheck the pull image secret every 24 hours(1d).
+	PullImageSecretRecheckPeriod *metav1.Duration `json:"pullImageSecretRecheckPeriod"`
```

Comment:
Is this the only config surface? I'm surprised we're expecting to default this tightening on when the feature graduates to beta. I expected the other way around, default to 0 / off, but let admins opt in.

Reply:
sounds good! However bad the current behavior may be, we don't to break pods inadvertantly relying on it so opt in makes sense.

Reply:
I concur with a note:

The reasoning was as follows:

Currently admins desiring to be secure by default in a multi user or reduction of permission to access an image scenario, were forced into restricting away the if not present policy by employing the AlwaysPullImages admission controller https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#alwayspullimages ...
The admins could also require `:latest tags`.. or employ a scheme to gc images.

Thus when we make the default off we will be asking admins to "trust" IfNotPresent now processing only if you they explicitly enable secure IfNotPresent behavior.

Reply:
Yes, this will be a behavior change when it promoted to beta if we default it to 24h. I think we wanted to make it a secure way when we discussed about the default behavior here. I agree we might not break current users who is using `IfNotPresent` and this new behavior may caused that the pod crash after this recheck.

So we will change the default value to 0. It means that we don't recheck the secret again for `IfNotPresent` ImagePullPolicy.
```
	// if the pullImageSecretRecheckPeriod is 0, we don't need to check the secret again
	if m.pullImageSecretRecheckPeriod.Duration == 0 {
		return pulledBySecret, true, nil
	}
```

I will update the PR.

Reply:
> Thus when we make the default off we will be asking admins to "trust" IfNotPresent processing only if they explicitly enable secure IfNotPresent behavior.

BTW, can we add a note that suggests user to use a recheck period to be more secure?

Reply:
default change and comment lgtm. traced through to where this is used and had a question about whether the *re*validation duration is what we want to decide whether to *ever* validate a pull secret